sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/constraints.scala

```diff
@@ -198,3 +268,30 @@ case class ForeignKeyConstraint(
     copy(userProvidedCharacteristic = c)
   }
 }
+
+
+/** This class is used to represent a collection of constraints in a table.
+ * It is used for analysis purposes only.
+ *
+ * @param children The list of constraint expressions
+ */
+case class Constraints(children: Seq[Expression]) extends Expression with Unevaluable {
```

Comment:
why can't we use `Seq[TableConstraint]` here?

Reply:
BTW why do we need this container? can't we directly put  `Seq[TableConstraint]` into `TableSpec`?

Reply:
> BTW why do we need this container? can't we directly put Seq[TableConstraint] into TableSpec?

The implementation of `UnresolvedTableSpec.withNewChildrenInternal` would be a bit ugly...but you are right, having the container seems more complicated